src/main/java/org/opensearch/ad/feature/SearchFeatureDao.java

```diff
@@ -114,9 +116,15 @@ public SearchFeatureDao(
         this.interpolator = interpolator;
         this.clientUtil = clientUtil;
         this.maxEntitiesForPreview = maxEntitiesForPreview;
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(MAX_ENTITIES_FOR_PREVIEW, it -> this.maxEntitiesForPreview = it);
         this.pageSize = pageSize;
-        clusterService.getClusterSettings().addSettingsUpdateConsumer(PAGE_SIZE, it -> this.pageSize = it);
+        try {
+            Map<Setting<?>, Consumer<?>> settingsUpdateConsumers = new HashMap<>();
+            settingsUpdateConsumers.put(MAX_ENTITIES_FOR_PREVIEW, it -> this.maxEntitiesForPreview = (int) it);
+            settingsUpdateConsumers.put(PAGE_SIZE, it -> this.pageSize = (int) it);
+            clusterService.getClusterSettings().addSettingsUpdateConsumer(settingsUpdateConsumers);
+        } catch (Exception e) {
+            // TODO Handle this
```

Comment:
[handleAddSettingsUpdateConsumerRequest](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/extensions/AddSettingsUpdateConsumerRequestHandler.java#L60) throws a generic exception here. Can we create an issue to make it more specific so as to handle it here later?

Reply:
Yeah, my TODO was kind of thinking it's better to handle these upstream, I just haven't gotten that far to thinking about how.  I'll make an issue.